### ISSUE TITLE
[Feature request] Possibility to add annotations to pods on Kubernetes

### ISSUE THREAD

**Issue by Emelieh21** [openanalytics/shinyproxy#152]
Hey guys,

I was wondering if the possibility to add [annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/) to pods launched by shinyproxy on Kubernetes could be added? This would enable us to [attach an AWS IAM role to a container](//github.com/jtblin/kube2iam#kubernetes-annotation) and use AWS resources in shiny apps.

Many thanks in advance! And thanks for all the great work.

Cheers,

Emelie

**Comment by lukeplausin** [comment]
I would also like to add annotations to pods launched by shiny proxy for exactly the same reason. I tried to alter the code to this end but my Java skills aren't good enough.

If anyone is still maintaining the project, this would be a really useful feature!

**Comment by lukeplausin** [comment]
In case anyone else is having the same issue, I found a workaround.
Shiny proxy is launching pods into a dedicated namespace. I launched another sidecar container which constantly annotates any pods which don't already have annotation.

(in `spec` section of deployment):

```
- name: shiny-role-spammer
  image: zlabjp/kubernetes-resource
  command: ["/bin/bash", "-c", "while [ 1 ]; do kubectl -n external annotate pods --all iam.amazonaws.com/role=\"arn:aws:iam::000245656003:role/shiny-external\"; sleep 2; done"]
```

**Comment by RogerioScussel** [comment]
good idea! Perfect!

**Comment by LEDfan** [comment]
Hi all

Thank you for your PR!
In order to not have to implement every Kubernetes feature in ContainerProxy, we decided to add a generic way of adding Kubernetes specific configuration to a ContainerProxy app. This feature is called `kubernetes-pod-patches` and allows to *patch* the specification of the Pod before it is created by ContainerProxy.
More information about this feature can be found [here](https://shinyproxy.io/documentation/configuration/#apps) (you have to scroll down a bit).
This should cover your use-case. Feel free to open an issue if you experience any issue with this feature.
The change is included in the just [ShinyProxy 2.4.0](https://shinyproxy.io/downloads/#240)

Again thanks for your contribution.

**Comment by Emelieh21** [comment]
Thanks @LEDfan this solution is really great - just see it now!

After some trial and error I managed to get it to work for adding IAM role annotations in the following way:

``` 
  - id: 01_hello
    displayName: Hello Application
    description: Application which demonstrates the basics of a Shiny app
    containerCmd: ["R", "-e", "shinyproxy::run_01_hello()"]
    containerImage: openanalytics/shinyproxy-demo
    kubernetes-pod-patches: |
     - op: add
       path: /metadata/annotations
       value: { "iam.amazonaws.com/role": "<YOUR-IAM-ROLE>" }
```
